backend/backendutil/search.go

```diff
@@ -66,12 +79,12 @@ func Match(e *message.Entity, seqNum, uid uint32, date time.Time, flags []string
 		if err != nil {
 			return false, err
 		}
-		t = t.Round(24 * time.Hour)
+		t = time.Date(t.Year(), t.Month(), t.Day(), 0, 0, 0, 0, time.UTC)
 
 		if !c.SentBefore.IsZero() && !t.Before(c.SentBefore) {
 			return false, nil
 		}
-		if !c.SentSince.IsZero() && !t.After(c.SentSince) {
+		if !c.SentSince.IsZero() && !t.After(c.SentSince) && !t.Equal(c.SentSince) {
```

Comment:
Ditto: `Time.After` checks for inequality

Reply:
Oh, you are right on Before but for After there is more: RFC 3501 names it "since" as it is >= (name in go-imap is somewhat misleading). In this particular case we skip the message if message matches the inverted criteria. But !t.After is <= (inverse of >), while we want inverse of >= which is <, therefore additional condition is needed to make <= into <.

Reply:
Ah. Can we use `t.Before` instead of `!t.After` then maybe?

Reply:
Right, we should (how come i find the least clean solution to the problem every single time?)